src/ImageCropper/ImagePreview.tsx

```diff
@@ -60,34 +37,50 @@ export class ImagePreview extends React.Component<ImagePreviewProps, ImagePrevie
                 </g>
             </svg>
         );
-    }
+    }, []);
 
-    render() {
-        return (
-            <div className={"profile-image-container " + this.props.previewClassName}>
-                <div className="profile-image">
-                    {this.state.cropDataResult && <img src={this.state.cropDataResult} alt="cropped image" id="profile-image" />}
-                    {!this.state.cropDataResult && this.defaultProfileSvg}
-                </div>
-                <input
-                    ref={(file) => {
-                        this.fileInput = file;
-                    }}
-                    type="file"
-                    id="fileInput"
-                    name="profileImage"
-                    accept="image/*"
-                    onChange={this.props.handleUploadImage}
-                    onClick={(e: any) => {
-                        if (e.target && e.target.value) {
-                            e.target.value = null;
-                        }
-                    }}
-                />
-                <button type="button" className="btn btn-primary custom-button" onClick={this.onFileInputClick}>
-                    <span>{this.props.selectButtonText || "Select Image"}</span>
-                </button>
-            </div>
-        );
-    }
-}
+    /**
+     * handle file click or select file click
+     */
+    const onFileInputClick = React.useCallback(
+        (e: React.ChangeEvent<HTMLInputElement> & React.DragEvent<HTMLInputElement>) => {
+            const { previewSrc, handleUploadImage } = props;
+            if (previewSrc) {
+                handleUploadImage(e, previewSrc);
+            } else {
+                fileInput.current.click();
+            }
+        },
+        [fileInput]
+    );
+
+    React.useEffect(() => {
+        if (props.cropResult) {
+            setCropDataResult(props.cropResult);
+        } else if (props.previewSrc) {
+            setCropDataResult(props.previewSrc);
+        }
+    }, [props.cropResult, props.previewSrc]);
+
+    return (
+        <div className={"profile-image-container " + props.previewClassName}>
+            <div className="profile-image">{cropDataResult ? <img src={cropDataResult} alt="cropped image" id="profile-image" /> : defaultProfileSvg}</div>
+            <input
+                ref={fileInput}
+                type="file"
+                id="fileInput"
+                name="profileImage"
+                accept="image/*"
+                onChange={props.handleUploadImage}
+                onClick={(e: React.MouseEvent<HTMLInputElement, MouseEvent>) => {
+                    if (e.target && (e.target as any).value) {
+                        e.target["value"] = null;
+                    }
+                }}
```

Comment:
You don't need to verify if `e.target` exist. It's like verifying if React as a library is working as intended. Additionally, why are you setting the value from here? That's direct DOM manipulation

Reply:
I cannot remember what is it. But last time we added this to address a specific issue. It might be related to drag and drop image, just a guess though. Also, this field is hidden. It isn't shown, we utilized the change event of the field instead.